src/praisonai-agents/praisonaiagents/tools/edit_tools.py

```
@@ -423,6 +450,121 @@ def _run_diagnostics(self, safe_path: str, display_path: str) -> str:
+        level and swallowed: a formatter must never fail an edit.
+        """
+        if self._post_edit_format == "off":
+            return
+        try:
+            resolved = self._format_command(safe_path)
+            if resolved is None:
+                return
+            _tool_name, argv = resolved
+
+            import subprocess
+            subprocess.run(
```

Ruff (0.15.18)

[error] `subprocess` call: check for execution of untrusted input

(S603)